@here/harp-mapview/test/stubElevationProvider.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { GeoCoordinates, mercatorTilingScheme, TileKey, TilingScheme } from "@here/harp-geoutils";
+import * as THREE from "three";
+import { DisplacementMap, TileDisplacementMap } from "../lib/DisplacementMap";
+import { ElevationProvider } from "../lib/ElevationProvider";
+
+class ElevationProviderFake implements ElevationProvider {
+    getHeight(_geoPoint: GeoCoordinates, _level?: number): number | undefined {
+        return undefined;
+    }
+
+    sampleHeight(_geoPoint: GeoCoordinates, _tileDisplacementMap: TileDisplacementMap): number {
+        return 0;
+    }
+
+    rayCast(_x: number, _y: number): THREE.Vector3 | undefined {
+        return undefined;
+    }
+
+    getDisplacementMap(_tileKey: TileKey): TileDisplacementMap | undefined {
+        return undefined;
+    }
+
+    getTilingScheme(): TilingScheme | undefined {
+        return undefined;
+    }
+
+    clearCache(): void {
+        return;
+    }
+}
+
+export type ElevationProviderStub = sinon.SinonStubbedInstance<ElevationProvider>;
+
+/**
+ * Creates an ElevationProvider stub.
+ * @param sandbox Sinon sandbox to keep track of created stubs.
+ * @returns ElevationProvider stub.
+ */
+export function stubElevationProvider(sandbox: sinon.SinonSandbox): ElevationProviderStub {
+    const stub = sandbox.createStubInstance(ElevationProviderFake);
+    stub.getHeight.callsFake(geoPoint => {
```

Comment:
do you need this, why not provide a fake implementation in the ElevationProviderFake?

Reply:
Returning a stub is more flexible, it allows users to overwrite the behaviour with whatever they want.
Writing a fake with hardcoded behaviour would only be useful for my test.
In theory I wouldn't have to write the Fake class at all, but sadly sinon doesn't let you stub an interface directly, only classes.

Reply:
besides, getDisplacementMap stub is configured in the test to return different results depending on the passed tile key (see my answer below).

Reply:
getDisplacementMap makes sense, and it makes sense to return a stub, I'm more interested to know why you need to have callsFake for getHeight and sampleHeight, couldn't they just be in the fake class?

Reply:
createStubInstance() creates a fully stubbed instance of the ElevationProviderFake, so none of the method implementations are preserved. That's why I set the behaviour later with callsFake().